dbms/src/Client/Benchmark.cpp

```diff
@@ -183,7 +186,7 @@ class Benchmark
 		Stopwatch watch;
 
 		/// В цикле, кладём все запросы в очередь.
-		for (size_t i = 0; !interrupt_listener.check(); ++i)
+		for (size_t i = 0; !interrupt_listener.check() && (!(num_repetions > 0) || i < num_repetions); ++i)
```

Comment:
According to next three lines, you cannot specify `num_repetitions` greater than `queries.size()`.